scripts/lib/format-converter.ts

```diff
@@ -224,3 +230,93 @@ export function googleSheetToFlatData(
   }
   return records;
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type StringKeyed = { [index: string]: any };
+
+export function nestedToFlat(input: StringKeyed): StringKeyed {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const output: { [index: string]: any } = {};
+  for (const [fieldName, field] of Object.entries(input)) {
+    if (field === undefined) continue;
+    if (Array.isArray(field)) {
+      output[fieldName] = field.join(', ');
+    } else if (typeof field === 'object' && !Array.isArray(field)) {
+      for (const [nestedKey, nestedVal] of Object.entries(
+        nestedToFlat(field),
+      )) {
+        output[`${fieldName}.${nestedKey}`] = nestedVal;
+      }
+    } else {
+      output[fieldName] = field;
+    }
+  }
+  return output;
+}
+
+export function collectedIncentiveToGoogleSheet(
+  incentives: CollectedIncentive[],
+  fieldMappings: AliasMap,
+): sheets_v4.Schema$Sheet {
+  const standardizer = new SpreadsheetStandardizer(
+    FIELD_MAPPINGS,
+    VALUE_MAPPINGS,
+    false,
+  );
+
+  const headers = Object.values(fieldMappings).map(mapping => mapping[0]);
+  let rowData: sheets_v4.Schema$RowData[] = [];
+  incentives.forEach(incentive => {
+    const flattened = nestedToFlat(incentive);
+    const aliased = standardizer.convertToAliases(flattened);
+    // First ensure we have a header column for every value, including those
+    // not in our schema.
+    for (const key in aliased) {
+      if (!headers.includes(key)) {
```

Comment:
This seemed weird to me -- it seemed like it shouldn't be possible -- which is what led me to the comment in `convertToAliased`. I might not have understood all of what's going on here.

Reply:
Ah, OK.

It occurs to me, though, that the word/character count columns will be troublesome because they're formulas. If the sheet-to-JSON process just captures the displayed text, then JSON-to-sheet will render them as "hardcoded" values that don't change when the description does. On the whole I don't think it's the worst problem, but something to be aware of.